### PR TITLE
feat(InlineEdit): create InlineEdit component for React

### LINKED ISSUE #10003: Consume core updates: inline edit (Table) ( Follow up to issue)
Follow up to core issue https://github.com/patternfly/patternfly/issues/5736

### PR DESCRIPTION
**Pull request by adamviktora**
#10003 can be closed, this PR is just a nice to have and not a priority right now.

**Additional issues**: after creating `InlineEdit` component, we can finish this Table with editable rows demo / example: https://github.com/patternfly/patternfly-react/issues/6597, or we can create that examples without using InlineEdit component.

### CODE REVIEW

File: packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx
Comment (thatblindgeye) [inline packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx:6]: Maybe a question for @lboehling , but do we want to allow passing custom icons?

File: packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx
Comment (thatblindgeye) [inline packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx:40]: We should have a prop for the aria-label

File: packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx
Comment (thatblindgeye) [inline packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx:50]: Same as above re: aria-label

File: packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx
Comment (thatblindgeye) [inline packages/react-core/src/components/InlineEdit/InlineEditActionGroup.tsx:43]: Nit: can we make the saveButtonText default "Save", rather than adding onto the ternary here? Also not opposed to checking if `saveButtonProps.children` exists, but personally not sure if we need it.

File: packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx
Comment (thatblindgeye) [inline packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx:16]: Let's add a prop for the aria-label.

File: packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx
Comment (thatblindgeye) [inline packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx:17]: @lboehling similar question as the save/cancel button icons: do we want to support custom icon here? And do we also want to allow either using text instead ("Edit") or using both text and icon as seen in the [full page edit design guidelines](https://www.patternfly.org/components/inline-edit/design-guidelines#full-page-edit)?

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Some initial comments/questions below. It'd also be worth making a followup to add more examples/demos that are shown in the design guidelines, if we still intend to support those implementations.

File: packages/react-core/src/components/InlineEdit/InlineEdit.tsx
Comment (mcoker) [reply packages/react-core/src/components/InlineEdit/InlineEdit.tsx:27]: @thatblindgeye good question - it looks like the core examples aren't consistent, either. My guess is the inputs/buttons need a wrapper, and a `<form>` is a semantic wrapper for the inputs/buttons, so it's an option, and may be better than something generic like a `<div>` for those examples - especially for a core user that may not be using react and might handle form submissions in a more traditional/server-side way. And AFAIK it will also wire up hitting enter to submit the form, though the event handler will need to stop the form from submitting and refreshing the page.

I guess it's a valid _option_ for wrapping the editable thing/buttons, though as a default or in what context in a react app, I'd probably defer to y'all. I'd be down to chat about it if you'd like.

**Comment by thatblindgeye** [comment]
Made updates from most of my comments above along with other updates from the original draft. Wasn't sure exactly how customizable we'd want parts of this component to be, so currently exporting the InlineEditToggle and InlineEditActionGroup that can be customized a bit further, otherwise a consumer can just pass props to the base InlineEdit component. Also marked the component as beta so we're not locked into this particular API.

A TODO is adding unit/integration tests, which can be done once an API is approved or as a followup to this.

Also looks like there may be a styling bug for the save/cancel icons. In Core the example where they're standalone shows a gap:

Inline edit action group standalone with gap

But in context, there's no gap:

Inline edit action group in context without gap

File: packages/react-core/src/components/InlineEdit/InlineEdit.tsx
Comment (thatblindgeye) [reply packages/react-core/src/components/InlineEdit/InlineEdit.tsx]: For now I added a prop to customize the wrapper element, defaulting to "form". I had been thinking that within the context of a table row we may want a `tr` wrapper for semantics, maybe some other element in some other context, etc.

File: packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx
Comment (adamviktora) [inline packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx:14]: I would remove the "must be passed in", because there will always be a default 'Enable edit mode' `ariaLabel`.

File: packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx
Comment (adamviktora) [inline packages/react-core/src/components/InlineEdit/InlineEditToggle.tsx:41]: Why is aria-label `undefined` when children is passed in? Is it because when there is a readable text in the button we don't need an aria-label?

**Comment by thatblindgeye** [comment]
Per discussion we don't plan to implement a React inline edit component at this time, so will be closing this PR.

**Pull request closed without merging**
